File: src/main/java/me/jameschan/burrow/kernel/command/entry/EntriesCommand.java
package me.jameschan.burrow.kernel.command.entry;

import java.util.Arrays;
import java.util.List;
import me.jameschan.burrow.kernel.command.Command;
import me.jameschan.burrow.kernel.common.ExitCode;
import me.jameschan.burrow.kernel.context.RequestContext;
import picocli.CommandLine;

@CommandLine.Command(name = "entries", description = "Retrieve a list of entries.")
public class EntriesCommand extends Command {
  @CommandLine.Parameters(index = "0", description = "A list of IDs to retrieve.")
  private String idList;

  public EntriesCommand(final RequestContext context) {
    super(context);
  }

  @Override
  public Integer call() throws Exception {
    final var idListString = idList.trim();
    if (idListString.length() < 2) {
      buffer.append("Invalid id list string: ").append(idListString);
      return ExitCode.ERROR;
    }

    final var hoard = context.getHoard();
    final var formatter = context.getFormatter();
    final List<String> entryStringList =
        Arrays.stream(idListString.substring(1, idListString.length() - 1).split(","))
            .map(String::trim)
            .map(Integer::parseInt)
            .map(hoard::getById)
            .map(formatter::format)
            .toList();
    buffer.append(formatter.format(entryStringList));

    return ExitCode.SUCCESS;
  }
}

File: src/main/kotlin/carton/hoard/command/EntriesCommand.kt
package burrow.carton.hoard.command

import burrow.carton.hoard.Hoard
import burrow.carton.hoard.printer.EntriesPrinter
import burrow.carton.hoard.printer.EntriesPrinterContext
import burrow.kernel.terminal.*

@BurrowCommand(
    name = "entries",
    header = ["Displays specified entries."],
)
class EntriesCommand(data: CommandData) : Command(data) {
    @Parameters(
        index = "0..*",
        description = ["A list of IDs of entries to retrieve."]
    )
    private var entryIds: Array<Int> = arrayOf()

    override fun call(): Int {
        val storage = use(Hoard::class).storage
        val entries = entryIds
            .map { storage[it] }
            .map { it.id to storage.formatStore(it) }
            .toList()
        EntriesPrinter(stdout, EntriesPrinterContext(entries)).print()

        return ExitCode.OK
    }
}